cmd/esc/cli/client/client.go

```diff
@@ -560,7 +560,7 @@ func (pc *client) UpdateEnvironmentWithRevision(
 	})
 	if err != nil {
 		var diags *EnvironmentErrorResponse
-		if errors.As(err, &diags) && diags.Code == http.StatusBadRequest && len(diags.Diagnostics) != 0 {
+		if errors.As(err, &diags) && (diags.Code == http.StatusBadRequest || len(diags.Diagnostics) != 0) {
```

Comment:
This is because on certain 4xx errors, the pulumi service returns an object without {..., `"code": 4xx`}

Reply:
Hmm, should we fix this in PS? Sounds like it's not following a required API contract?

Reply:
I initially considered this but I think it's a much bigger lift - we follow this "pattern" in many routes where certain errors will return `code` and `message` by default, but otherwise not.

I think perhaps a better fix is to check the http response status code instead. @pgavlin I notice that we [don't return](https://github.com/pulumi/esc/blob/0a8fc4100f4cce21f9b45faa9ef379d100718696/cmd/esc/cli/client/client.go#L1156-L1174) the http response on 4xx/5xx errors, is this intentional?